frontend/src/components/cases/case-attachments-section.tsx

```diff
@@ -0,0 +1,735 @@
+"use client"
+
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query"
+import { formatDistanceToNow } from "date-fns"
+import {
+  AlertCircle,
+  Download,
+  Eye,
+  FileIcon,
+  FileSpreadsheet,
+  FileText,
+  ImageIcon,
+  Music,
+  Plus,
+  Presentation,
+  Trash2,
+  Video,
+} from "lucide-react"
+import Image from "next/image"
+import { useCallback, useRef, useState } from "react"
+import type { ApiError, CaseAttachmentRead } from "@/client"
+import {
+  casesCreateAttachment,
+  casesDeleteAttachment,
+  casesDownloadAttachment,
+  casesListAttachments,
+} from "@/client"
+import { Button } from "@/components/ui/button"
+import { Dialog, DialogContent } from "@/components/ui/dialog"
+import { Skeleton } from "@/components/ui/skeleton"
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@/components/ui/tooltip"
+import { toast } from "@/components/ui/use-toast"
+import { cn } from "@/lib/utils"
+
+interface CaseAttachmentsSectionProps {
+  caseId: string
+  workspaceId: string
+}
+
+interface CaseAttachmentDownloadResponse {
+  download_url: string
+  file_name: string
+  content_type: string
+}
+
+// Type definitions for API error responses
+interface ApiErrorDetail {
```

Comment:
see my other comment at the router. these types can be autogenerated through openapi